src/middleware/overview.middleware.js

```
@@ -55,6 +56,130 @@ const fetchEntityCounts = async (req, res, next) => {
+
+    const { deadlineDate, lastYearDeadline, twoYearsAgoDeadline } = getDeadlineHistory(dataset.deadline)
+
+    if (!deadlineDate || !lastYearDeadline || !twoYearsAgoDeadline) {
+      logger.error(`Invalid deadline dates for dataset: ${dataset.dataset}`)
+      return { dataset: dataset.dataset, dueNotice: false, overdueNotice: false, deadline: undefined }
+    }
+
+    if (resource) {
+      const startDate = new Date(resource.startDate)
+
+      if (!resource.startDate || isNaN(startDate.getTime())) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)